dartagnan/src/main/java/com/dat3m/dartagnan/parsers/program/visitors/VisitorLitmusOpenCL.java

```diff
@@ -0,0 +1,280 @@
+package com.dat3m.dartagnan.parsers.program.visitors;
+
+import com.dat3m.dartagnan.configuration.Arch;
+import com.dat3m.dartagnan.exception.ParsingException;
+import com.dat3m.dartagnan.expression.Expression;
+import com.dat3m.dartagnan.expression.ExpressionFactory;
+import com.dat3m.dartagnan.expression.IConst;
+import com.dat3m.dartagnan.expression.op.IOpBin;
+import com.dat3m.dartagnan.expression.type.IntegerType;
+import com.dat3m.dartagnan.expression.type.TypeFactory;
+import com.dat3m.dartagnan.parsers.LitmusOpenCLBaseVisitor;
+import com.dat3m.dartagnan.parsers.LitmusOpenCLParser;
+import com.dat3m.dartagnan.parsers.program.utils.AssertionHelper;
+import com.dat3m.dartagnan.parsers.program.utils.ProgramBuilder;
+import com.dat3m.dartagnan.program.Program;
+import com.dat3m.dartagnan.program.Register;
+import com.dat3m.dartagnan.program.event.EventFactory;
+import com.dat3m.dartagnan.program.event.Tag;
+import com.dat3m.dartagnan.program.event.arch.opencl.OpenCLAtomCAX;
+import com.dat3m.dartagnan.program.event.arch.opencl.OpenCLAtomOp;
+import com.dat3m.dartagnan.program.event.core.Event;
+import com.dat3m.dartagnan.program.event.core.Label;
+import com.dat3m.dartagnan.program.event.core.Load;
+import com.dat3m.dartagnan.program.event.core.Store;
+import com.dat3m.dartagnan.program.memory.MemoryObject;
+import org.antlr.v4.runtime.misc.Interval;
+
+public class VisitorLitmusOpenCL extends LitmusOpenCLBaseVisitor<Object> {
```

Comment:
Why is this class still here? Now the changes should be implemented on top of the LitmusC visitor

Reply:
Removed